pipeline/metadata/satellite.py

```diff
@@ -612,28 +623,15 @@ def partition_satellite_input(
         "Bad input number of partitions; always use NUM_SATELLITE_INPUT_PARTITIONS."
     )
 
-  filename = pathlib.PurePosixPath(line[0]).name
-  if '.gz' in pathlib.PurePosixPath(filename).suffixes:
-    filename = pathlib.PurePosixPath(filename).stem
-
-  if filename in [
-      flatten_satellite.SATELLITE_RESOLVERS_FILE,
-      flatten_satellite.SATELLITE_TAGGED_RESOLVERS_FILE,
-      flatten_satellite.SATELLITE_TAGGED_ANSWERS_FILE,
-      flatten_satellite.SATELLITE_TAGGED_RESPONSES
-  ]:
+  filename = _get_filename(line[0])
+  if filename in flatten_satellite.SATELLITE_ANSWER_TAG_FILES:
     return 0
-  if filename == flatten_satellite.SATELLITE_BLOCKPAGES_FILE:
+  if filename in flatten_satellite.SATELLITE_RESOLVER_TAG_FILES:
     return 1
-  if filename in [
-      flatten_satellite.SATELLITE_INTERFERENCE_FILE,
-      flatten_satellite.SATELLITE_INTERFERENCE_ERR_FILE,
-      flatten_satellite.SATELLITE_ANSWERS_CONTROL_FILE,
-      flatten_satellite.SATELLITE_RESPONSES_CONTROL_FILE,
-      flatten_satellite.SATELLITE_RESULTS_FILE,
-      flatten_satellite.SATELLITE_ANSWERS_ERR_FILE
-  ]:
+  if filename in flatten_satellite.SATELLITE_BLOCKPAGE_FILES:
```

Comment:
Make it clearer that there's only one file:
```suggestion
  if filename == flatten_satellite.SATELLITE_BLOCKPAGE_FILE:
```

In fact, I preferred when the list was inline here, to it was easier to tell how much work needed to be done for each partition to make them separate inputs.

Reply:
It's not as visible in the diff view. But the version of this function with the lists factored out instead of inlined is really much nicer to read.

![A5Ji3MiGKL4wHPH](https://user-images.githubusercontent.com/1127209/156818821-40106f60-d70a-4053-84b1-638a69d4e64e.png)

Probably the file lists should be in satellite.py instead of flatten_satellite.py though. I have a future refactoring planned where the filename-conditioning in flatten_satelllite.py also gets pulled up the pipeline to partition much earlier. That should let all the filenames come live here.

Reply:
removed the single-element blockpage file list